### ISSUE TITLE
Implementation of CGD multicost

### ISSUE THREAD

**Issue by allaffa** [allaffa/CompetitiveGradientDescent#39]
An implementation of Competitive Gradient Descent with multiple loss functions, one per agent.
